.github/workflows/release_win_freethreading.yml

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) ONNX Project Contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+name: WindowsRelease (freethreading)
+
+on:  # Specifies the event triggering the workflow
+  workflow_call:  # Indicates that this is a reusable workflow
+    inputs:
+      os:
+        required: true
+        type: string
+        
+permissions:  # set top-level default permissions as security best practice
+  contents: read
+
+jobs:
+  build:
+    if: github.event_name != 'pull_request' || startsWith( github.base_ref, 'rel-') || contains( github.event.pull_request.labels.*.name, 'run release CIs')
+    runs-on: windows-latest
+    strategy:
+      matrix:
+        python-version: ['3.13t']
+        architecture: ['x64']
+    steps:
+    - name: Checkout ONNX
+      uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      with:
+         submodules: 'recursive'
+         persist-credentials: false
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: Quansight-Labs/setup-python@v5
+      with:
+        python-version: ${{ matrix.python-version }}
+        architecture: ${{ matrix.architecture }}
+
+    - name: Add msbuild to PATH
+      uses: microsoft/setup-msbuild@6fb02220983dee41ce7ae257b6f4d8f9bf5ed4ce # v2.0.0
+      with:
+        msbuild-architecture: ${{ matrix.architecture }}
+
+    - name: Install Python dependencies
+      run: |
+        python -m pip install -q --upgrade pip
+        python -m pip install -q -r requirements-release_freethreading_win.txt
+
+    - name: Build ONNX wheel
+      run: |
+        $arch = 'x64'
+        if ('${{ matrix.architecture }}' -eq 'x86') {
+          $arch = 'Win32'
+        }
+
+        . .\workflow_scripts\protobuf\build_protobuf_win.ps1 -arch $arch
+
+
+        echo "Install ONNX"
+        $Env:ONNX_ML=1
+        $Env:CMAKE_ARGS="-DONNX_USE_PROTOBUF_SHARED_LIBS=OFF -DONNX_USE_LITE_PROTO=ON"
+
+        if ('${{ github.event_name }}' -eq 'schedule' -or '${{ github.event_name }}' -eq 'workflow_dispatch') {
+          echo "Build preview build PyPI package"
+          (Get-Content -Path 'pyproject.toml') | ForEach-Object { $_ -replace 'name = "onnx"', 'name = "onnx-weekly"' } | Set-Content -Path 'pyproject.toml'
+          $Env:ONNX_PREVIEW_BUILD=1
+        }
+        python -m build --wheel
+        Get-ChildItem -Path dist/*.whl | foreach {python -m pip install --upgrade $_.fullname}
+ 
+    - name: Test the installed wheel
+      run: |
+        
+        # 2025.03.06 jupyter_client and ipython have conflicting dependencies
+        # therefore [tool.pytest.ini_options] in pyproject.toml is overwritten
+        # because it contains nbval which cannot be installed
+
+        echo "[pytest]" > pytest.ini
+        echo "addopts = --tb=short --color=yes" >> pytest.ini
+        echo "testpaths =" >> pytest.ini
+        echo "    onnx/test" >> pytest.ini
+        echo "    onnx/examples" >> pytest.ini
+        python -m pip install -v pillow
+        pytest
```

Comment:
Is it possible to fix pyproject.toml or nbval etc.?

Reply:
I have not found a way to store python version specific configuration that I like. (That's not to say that there aren't any). 

Free threading is something new, so I expect that the situation will improve in the near future and that some “hacks” or “duplicate” files can be removed. 

My experiments went in the direction of installing suitable dependencies. For example, installing the latest versions of ipython, jupyter_client and nbval directly from the github repo. My preference would be to fix the problem in the original project(s), although I haven't found out exactly where it should be fixed.